src/repositories/timesheet/models.rs

```diff
@@ -0,0 +1,52 @@
+use chrono::{NaiveDate, NaiveDateTime};
+use sqlx::prelude::FromRow;
+use uuid::Uuid;
+
+#[derive(Debug, FromRow)]
+pub struct TimesheetDb {
+    pub id: Uuid,
+    pub start_date: NaiveDate,
+    pub end_date: NaiveDate,
+    pub total_hours: f32,
+    pub is_editable: bool,
+    pub manager_note: Option<String>,
+    // foreign keys
+    pub user_id: Uuid,
+    pub company_id: Uuid,
+    pub event_id: Uuid,
+    // timestamps
+    pub created_at: NaiveDateTime,
+    pub edited_at: NaiveDateTime,
+    pub deleted_at: Option<NaiveDateTime>,
+}
+
+#[derive(Debug, FromRow)]
+pub struct TimesheetCreateData {
+    pub start_date: NaiveDate,
+    pub end_date: NaiveDate,
+    pub total_hours: f32,
+    pub is_editable: bool,
+    pub manager_note: Option<String>,
+    // foreign keys
+    pub user_id: Uuid,
+    pub company_id: Uuid,
+    pub event_id: Uuid,
+}
+
+// add another struct if necessary, otherwise do not fucking duplicate
+#[allow(dead_code)]
+pub type TimesheetUpdateData = TimesheetCreateData;
```

Comment:
It doesn't make sense to edit a user ID for a timesheet

Thus this should be a separate type and have Option() arguments

(basically the same as is now repositories)

Reply:
Seconding this.